okcupyd/json_search.py

```diff
@@ -118,17 +167,539 @@ def __call__(self, response_dictionary):
+    def transform(status="any"):
+        status = status.lower()
+        if status == 'not single' or status == 'married':
+            return 'not_single'
+        elif status in ('single', 'any'):
+            return status
+        else:
+            raise TypeError
+
+
+class MonogamyFilter(search_filters.filter_class):
+    output_key = "monogamy"
+    acceptable_values = ("yes", True, "no", False, 'monogamous', 'non-monogamous', 'any')
+    # types = str
+    def transform(monogamy="unknown"):
+        # default is 'unknown'
+        if isinstance(monogamy, six.string_types):
+            monogamy = monogamy.lower()
+        if monogamy in ('yes', True, 'monogamous'):
+            return 'yes'
+        elif monogamy in ('no', False, 'non-monogamous'):
+            return 'no'
+        elif monogamy == 'any':
+            return 'unknown' # default
+        else:
+            raise TypeError
+
+
+class LookingForFilter(search_filters.filter_class):
+    output_key = "looking_for"
+    # description = "One or more relationship types sought that search results must match."
+    acceptable_values = (
+        'new friends', 'short-term', 'short-term dating',
+        'long-term', 'long-term dating', 'casual sex',
+        'sex', 'any')
+    types = "string or list of strings"
+    @util.makelist_decorator
+    def transform(kinds=[]):
+        result = []
+        for k in kinds:
+            k = k.lower()
+            if k == 'new friends':
+                result.add('new_friends')
+            elif k == 'short-term' or k == 'short-term dating':
+                result.add('short_term_dating')
+            elif k == 'long-term' or k == 'long-term dating':
+                result.add('long_term_dating')
+            elif k == 'casual sex' or k == 'sex':
+                result.add('casual_sex')
+            elif k != 'any':
+                raise TypeError
+            else: result.extend(('new_friends', 'short_term_dating', 'long_term_dating', 'casual_sex')) # or []
+        return result
+
+
+## old param keywords: A list or space-delimeted string of words to search for.
+## new param interests: A list of ints or str(int)
+class InterestsFilter(search_filters.filter_class):
+    output_key = "interest_ids"
+    description = "One or more interest ints that search results must match."
+    types = "int or intstring, or list of such"
+    @util.makelist_decorator
+    def transform(interests=[]):
+        for k in interests:
+            int(k) # verify that each element converts to an int
+        # expected to be list of bigint strings
+        return interests
+
+
+class EducationFilter(search_filters.filter_class):
+    output_key = "education"
+    # description = "One or more education levels that search results must match."
+    acceptable_values = (
+        'high school', 'two-year college', 'college', 'university', 'college/university',
+        'post-grad', 'masters program', 'law school', 'med school', 'ph.d program'
+        )
```

Comment:
Don't understand, you mean `'high school` should start in the same column as the `(` on the line above?